### ISSUE TITLE
forge: build compiler version inconsistency

### ISSUE THREAD

**Issue by dekz** [foundry-rs/foundry#8062]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (fbad377 2024-06-04T00:17:42.494712000Z)

### What command(s) is the bug in?

forge build

### Operating System

macOS (Apple Silicon)

### Describe the bug

Recently we noticed an inconsistency in behaviour between multiple `forge build` runs. We compile multiple contracts with diffrent solc version pragmas.

```
sol λ forge clean && forge build
[⠊] Compiling...
[⠒] Compiling 32 files with Solc 0.7.6
[⠢] Compiling 16 files with Solc 0.8.10
[⠆] Compiling 60 files with Solc 0.6.12
[⠰] Compiling 36 files with Solc 0.8.25
[⠒] Compiling 11 files with Solc 0.8.9
[⠘] Solc 0.8.9 finished in 162.87ms
[⠑] Solc 0.8.10 finished in 189.75ms
[⠰] Solc 0.7.6 finished in 1.03s
[⠒] Solc 0.8.25 finished in 1.73s
[⠆] Solc 0.6.12 finished in 3.38s
Compiler run successful with warnings:
Warning (2018): Function state mutability can be restricted to pure
...


sol λ forge clean && forge build
[⠊] Compiling...
[⠒] Compiling 60 files with Solc 0.6.12
[⠢] Compiling 51 files with Solc 0.8.25
[⠆] Compiling 16 files with Solc 0.8.10
[⠰] Compiling 13 files with Solc 0.8.9
[⠒] Compiling 15 files with Solc 0.7.6
[⠑] Solc 0.8.25 finished in 161.35ms
[⠃] Solc 0.8.9 finished in 166.43ms
[⠘] Solc 0.8.10 finished in 195.36ms
[⠒] Solc 0.7.6 finished in 1.02s
[⠢] Solc 0.6.12 finished in 3.39s

Error:
Compiler run failed:
Warning (2018): Function state mutability can be restricted to pure
...
Error (5333): Source file requires different compiler version (current compiler is 0.8.25+commit.b61c2a91.Darwin.appleclang) - note that nightly builds are considered to be strictly less than the released version
 --> lib/cryptoalgebra-algebra/src/core/contracts/libraries/FullMath.sol:2:1:
  |
2 | pragma solidity ^0.4.0 || ^0.5.0 || ^0.6.0 || ^0.7.0;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

Currently observing around a 50/50 flakiness on the previous nightly [`nightly-fbad377ab26a432e48444cf324feee1195a30960`](https://github.com/foundry-rs/foundry/releases/tag/nightly-fbad377ab26a432e48444cf324feee1195a30960).

Downgrading to [`nightly-0fc39763d46b8ffab5fa4eaeb2f65ae078fa07de`](https://github.com/foundry-rs/foundry/releases/tag/nightly-0fc39763d46b8ffab5fa4eaeb2f65ae078fa07de) appears to not contain this issue.

**Comment by klkvr** [comment]
Could you please provide a repro for this?

**Comment by klkvr** [comment]
Closed by #8061, feel free to reopen if issue persists
